### PR TITLE
Add canonical IMDb reducer baseline

### LINKED ISSUE #24: Canonical IMDb pipeline baseline
## Goal
Begin the typed IMDb pipeline rebuild from roadmap #13 in one meaningful Codex-sized slice: keep canonical IDs, fix the known cast/director extraction bugs, and produce a typed artifact without breaking the existing CLI flow.

## Why
The current reducer still has the original learning-era failure modes: director string slicing, actor-only cast filtering, duplicate-title grouping, quantile filtering, and CSV/list string round-tripping. This issue should establish the pipeline baseline that later TMDB, embeddings, and API work can build on.

## Scope
- Preserve IMDb `tconst` as a first-class output column and avoid grouping by title in a way that collapses duplicate titles.
- Replace director `x[:9]` slicing with proper comma splitting and keep up to the top 3 director IDs/names.
- Include both `actor` and `actress` categories from `title.principals.tsv`, keeping up to the top 5 cast members per movie in principal order when that ordering is available.
- Add an explicit vote-threshold path, defaulting to 1,000 votes, while keeping the existing percentage/CLI behavior compatible enough for current scripts/tests.
- Add typed output support, preferably Parquet when dependencies support it, without removing the existing CSV output path required by the current app/CLI.
- Add regression tests using small synthetic DataFrames/files for the known bugs; do not require downloading real IMDb data.

## Non-goals
- Do not add TMDB, sentence-transformers, MovieLens, Postgres, pgvector, FastAPI, Docker, or deployment work.
- Do not rewrite the recommender model or web UI.
- Do not remove the existing CSV-based app path until a later issue migrates consumers.

## Acceptance criteria
- Tests prove duplicate titles remain distinct by `tconst`.
- Tests prove actress principals are included in cast extraction.
- Tests prove multiple directors are split correctly instead of sliced.
- Tests prove list-like fields are available as typed Python lists before CSV serialization and/or in the typed artifact.
- The existing test suite passes via `make test`.
- Implementation report includes changed files, exact verification output, and any remaining migration notes.

## Verification
Run `make test` and include the real output.

### PR DESCRIPTION
**Pull request by hazy-dreams**
## Summary
- Preserve `tconst` through the reduced dataset and stop collapsing duplicate titles
- Split director IDs on commas, keep up to three director IDs/names, and retain the scalar `director` column for current consumers
- Include both `actor` and `actress` principals in cast extraction, preserving principal ordering up to five cast members
- Add explicit `--min-votes` CLI support with a 1,000-vote default
- Add typed list-valued reducer fields and best-effort Parquet output while preserving CSV output
- Add synthetic regression tests for duplicate titles, actress principals, multi-director splitting, typed list fields, and vote thresholding

Closes #24

## Verification
This branch is based on current `master`, where #22's `Makefile` is not merged yet, so I used the existing test command directly.

First, plain `python -m pytest -q` on the ambient shell failed because Django was not installed there:

```text
ModuleNotFoundError: No module named 'django'
```

Using the branch-local `.venv` created during implementation:

```bash
.venv/bin/python -m pytest -q
.........                                                                [100%]
9 passed in 2.58s
```

## Migration notes
- Current CSV-based app/CLI path is preserved.
- Parquet output is best-effort and skipped if pandas has no Parquet engine installed.
- The recommender still consumes the scalar `director` column; canonical list fields are now available for later migration work.

### CODE REVIEW

File: src/dataset_reducer.py [hunk 1]
```diff
@@ -14,15 +14,179 @@
...
+            "genres",
+            "score",
+            "cast_ids",
+            "actors",
+        ]
+        return (
+            metadata[columns]
+            .sort_values("score", ascending=False)
+            .reset_index(drop=True)
+        )
```
[the remark this answers is not in the document]
Comment (hazy-dreams) [reply src/dataset_reducer.py:183]: @codex address feedback

[the remark this answers is not in the document]
Comment (hazy-dreams) [reply src/dataset_reducer.py:183]: @codex where is the change you mentioned in your review-thread reply? You said you committed `9d589e7` and created a follow-up PR titled “Disambiguate duplicate movie titles for reducer consumers,” but I do not see a PR targeting `codex/24-canonical-imdb-pipeline-baseline`, and the current #27 diff does not include the described `src/movie_recommender.py` / duplicate-title disambiguation changes. Please either link the follow-up PR/branch or create/update the visible follow-up PR targeting this branch.
